src/ai/backend/agent/docker/kernel.py

```diff
@@ -312,7 +312,11 @@ async def list_files(self, container_path: str):
 
         files = []
         for f in os.scandir(sys.argv[1]):
-            fstat = f.stat()
+            if f.is_symlink():
+                fstat = os.lstat(f)
+            else:
+                fstat = f.stat()
```

Comment:
According to [the official docs](https://docs.python.org/3.12/library/os.html#os.DirEntry.stat), we can simpy write this:
```suggestion
            fstat = f.stat(follow_symlinks=False)
```